src/components/ProvisionSmartWalletDialog.tsx

```diff
@@ -0,0 +1,121 @@
+  smartWalletProvisionedAtom,
+} from 'store/app';
+import { querySwingsetParams } from 'utils/swingsetParams';
+import type { ChainStorageWatcher } from 'store/app';
+import ActionsDialog from './ActionsDialog';
+import { provisionSmartWallet } from 'service/wallet';
+
+// Increment every time the current terms change.
+export const currentTermsIndex = 1;
+
+const feeDenom = 10n ** 6n;
+
+const useSmartWalletFeeQuery = (watcher: ChainStorageWatcher | null) => {
+  const [smartWalletFee, setFee] = useState<bigint | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  useEffect(() => {
+    const fetchParams = async () => {
+      assert(watcher);
+      try {
+        const params = await querySwingsetParams(watcher.rpcAddr);
+        console.debug('swingset params', params);
+        setFee(BigInt(params.params.powerFlagFees[0].fee[0].amount));
+      } catch (e) {
+        setError(e as Error);
+      }
+    };
+
+    if (watcher) {
+      fetchParams();
+    }
+  }, [watcher]);
+
+  return { smartWalletFee, error };
+};
+
+const ProvisionSmartWalletDialog = () => {
+  const chainConnection = useAtomValue(chainConnectionAtom);
+  const watcher = useAtomValue(chainStorageWatcherAtom);
+  const [provisionToastId, setProvisionToastId] = useAtom(provisionToastIdAtom);
+  const smartWalletProvisionRequired = useRef(false);
+  const [isSmartWalletProvisioned] = useAtom(smartWalletProvisionedAtom);
+  const [shouldBeOpenIfFeeLoaded, setShouldBeOpenIfFeeLoaded] = useState(false);
+  const { smartWalletFee, error: smartWalletFeeError } =
+    useSmartWalletFeeQuery(watcher);
+
+  useEffect(() => {
+    if (
+      isSmartWalletProvisioned === false &&
+      !smartWalletProvisionRequired.current
+    ) {
+      smartWalletProvisionRequired.current = true;
+      if (smartWalletFeeError) {
+        console.error('Swingset params error', smartWalletFeeError);
+        toast.error('Error reading smart wallet provisioning fee from chain.');
+        return;
+      }
+      setShouldBeOpenIfFeeLoaded(true);
+    } else if (
+      isSmartWalletProvisioned &&
+      smartWalletProvisionRequired.current
+    ) {
+      smartWalletProvisionRequired.current = false;
+      setShouldBeOpenIfFeeLoaded(false);
+      if (provisionToastId) {
+        toast.dismiss(provisionToastId);
+        setProvisionToastId(undefined);
+      }
+      toast.success('Smart wallet successfully provisioned.');
+    }
+  }, [
+    isSmartWalletProvisioned,
+    provisionToastId,
+    setProvisionToastId,
+    smartWalletFeeError,
+    smartWalletFee,
+  ]);
+
+  const smartWalletFeeForDisplay = smartWalletFee
+    ? smartWalletFee / feeDenom + ' BLD'
+    : null;
+
+  const provision = () => {
+    assert(chainConnection);
+    provisionSmartWallet(chainConnection, setProvisionToastId);
+    setShouldBeOpenIfFeeLoaded(false);
+  };
+
+  const body = (
+    <span>
+      To interact with contracts on the Agoric chain, a smart wallet must be
```

Comment:
For future work: should we have a utility for this in ui-kit? Worth an issue imo

Reply:
Possibly for the provisioning fee, though I'm not sure how likely that flow is to change (i.e. if we implement automatic provisioning). We already have a utility for the provision status. I'm not sure if the whole dialog flow is worth making into a component depending on how dapps want to handle it